### PR TITLE
brace: Fix brace drawing code

### LINKED ISSUE #577: braces degenerate with high amplitudes
High aspect ratios of braces, that is high amplitudes with low span values, create some weird shapes.
Not sure if that is too much asked from cetz, but it just seemed to me that `cetz.decorations.braces` should draw nice braces in many situations. I am not really sure what I expect exactly, but I think that raising the amplitude should not drastically change the shape.
image

```typst
#import "@preview/cetz:0.2.2"

#set page(height: auto, width: auto)
#set text(15pt)

#cetz.canvas(debug: false, {
  import cetz.draw: *

  let y = 0
  for amp in range(1, 5) {
    for deg in range(4) {
      let start = (deg * 6, y)
      let pointi = 30deg * deg
      cetz.decorations.brace(
        start,
        (rel: (x: 5), to: start),
        amplitude: amp,
        pointiness: pointi,
        outer-pointiness: 90deg,
      )
      if amp == 1 {
        content((2.5 + deg * 6, -0.3), [pointiness: #pointi], anchor: "north")
      }
    }
    content((0, y + 0.3), [amplitude: #amp], anchor: "south")
    y += amp
  }
})
```

### PR DESCRIPTION
**Pull request by johannes-wolf**
Simplifies and improves brace drawing. Pointiness values are now clamped between 0 and 90 degree, as values outsides that range lead to strange looking braces.

Fixes #577.

### CODE REVIEW

File: src/lib/decorations/brace.typ [hunk 1]
```diff
@@ -57,145 +61,85 @@
...
-  ..style,
-) = {
-  // Validate coordinates
-  let t = (start, end).map(coordinate.resolve-system)
+#let brace(start, end, ..style, name: none) = group(name: name, ctx => {
+  assert.eq(style.pos().len(), 0,
+    message: "Brace takes no additional positional arugments.")
+
+  // Resolve all coordinates
+  let (ctx, start, end) = coordinate.resolve(ctx, start, end)
```
Comment (fenjalien) [inline src/lib/decorations/brace.typ:71]: assert and coordinate checks should be outside of the returned function

**Review by fenjalien — CHANGES_REQUESTED** [review]

File: src/lib/decorations/brace.typ (hunk 1 above)
Comment (johannes-wolf) [reply src/lib/decorations/brace.typ]: Fixed.

**Review by fenjalien — APPROVED** [review]

File: src/lib/decorations/brace.typ [hunk 2]
```diff
@@ -41,11 +43,13 @@
 ///   #show-parameter-block("amplitude", ("number"), [
 ///     Sets the height of the brace, from its baseline to its middle tip.], default: .5)
 ///   #show-parameter-block("pointiness", ("ratio", "angle"), [
-///     How pointy the spike should be. #0deg or `0%` for maximum pointiness, #90deg or `100%` for minimum.], default: 15deg)
+///     How pointy the spike should be. #0deg or `100%` for maximum pointiness, #90deg or `100%` for minimum.], default: 15deg)
```
Comment (raphCode) [inline src/lib/decorations/brace.typ:46]: This mentions 100% for maximum and minimum, is this correct?

Comment (johannes-wolf) [reply src/lib/decorations/brace.typ:46]: > This mentions 100% for maximum and minimum, is this correct?

Oops.

File: src/lib/decorations/brace.typ [hunk 3]
```diff
@@ -41,11 +43,13 @@
...
 ///   #show-parameter-block("outer-pointiness", ("ratio", "angle"), [
-///     How pointy the outer edges should be. #0deg or `0` for maximum pointiness (allowing for a smooth transition to a straight line), #90deg or `1` for minimum. Setting this to #auto will use the value set for `pointiness`.], default: 15deg)
-///   #show-parameter-block("content-offset", ("number","length"), [
+///     How pointy the outer edges should be. #0deg or `100%` for maximum pointiness (allowing for a smooth transition to a straight line), #90deg or `1` for minimum. Setting this to #auto will use the value set for `pointiness`.], default: 15deg)
```
Comment (raphCode) [inline src/lib/decorations/brace.typ:48]: Here is a similar change that makes little sense to me: 100% = maximum, 1 = minimum

**Pull request merged**